### PR TITLE
jira plugin changes

### PR DESCRIPTION
**Pull request by KseniaLL**
error checking in lira plugin

### CODE REVIEW

**Review by dnorth98 — APPROVED** [review]

File: project/plugins/jira/__init__.py [hunk 1]
```diff
@@ -27,19 +29,54 @@ def inviteUser(email,configMap,allPermissions, plugin_tag, name):
...
-        add=requests.post(url+'/rest/api/2/group/user?groupname='+group, auth=(user, password),headers=headers, data=data )
-
-    log = 'Jira: ' + email.split('@', 1)[0] + ' added to ' + plugin_tag + '\n'
-    instruction = inviteMessage(configMap, plugin_tag)
-    return getJsonResponse("Jira Server",email, log, instruction)
+    #create a user
+    create = requests.post(url + '/rest/api/2/user', headers=headers, auth=(user, password), data=data)
+    if create.status_code > 201:
+        if create.status_code == 400:
+            log = plugin_tag + ' error: User already exists. Coudl not create user ' + username
```
Comment (haggaret) [inline project/plugins/jira/__init__.py:36]: typo here

File: project/plugins/jira/__init__.py [hunk 2]
```diff
@@ -27,19 +29,54 @@ def inviteUser(email,configMap,allPermissions, plugin_tag, name):
...
+        elif create.status_code == 401:
+            log = plugin_tag + ' error: User ' + name + ' is not authenticated'
```
Comment (haggaret) [inline project/plugins/jira/__init__.py:38]: I'm not sure this error message is correct - don't you get a 401 when the user *making* the call is not authorized? In this case your error message says the user you are trying to create is not authenticated...

File: project/plugins/jira/__init__.py [hunk 3]
```diff
@@ -27,19 +29,54 @@ def inviteUser(email,configMap,allPermissions, plugin_tag, name):
...
+        groups = getCLIgroups(configMap, plugin_tag, allPermissions)
+        #add user to the group
+        for group in groups:
+            add=requests.post(url+'/rest/api/2/group/user?groupname='+group, auth=(user, password),headers=headers, data=data )
+
+        if add.status_code > 201:
+            if add.status_code == 400:
+                log = plugin_tag + ": user " + username+ " requested an empty group name or the user already belongs to the group"
+            elif add.status_code == 401:
+                log = plugin_tag + ": current user " + username + " is not authenticated"
```
Comment (haggaret) [inline project/plugins/jira/__init__.py:59]: Same comment as above, not sure the message is correct

File: project/plugins/jira/__init__.py [hunk 4]
```diff
@@ -56,17 +93,37 @@ def removeUser(email, configMap,allPermissions, plugin_tag):
...
+
+        #Deactivating users is not enabled in the Api, users will be removed from all groups instead
+        #https: // jira.atlassian.com / browse / JRASERVER - 44801
+        for group in groupList:
+            delete=requests.delete(url+'/rest/api/2/group/user?groupname='+group+'&username=' + username, headers=headers,auth=(user, password))
+            if delete.status_code > 200:
+                if delete.status_code == 400:
+                    log = plugin_tag + ": user " + username + " requested an empty group name"
+                elif delete.status_code == 401:
+                    log = plugin_tag + ": current user " + username + " is not authenticated"
```
Comment (haggaret) [inline project/plugins/jira/__init__.py:114]: same comment as above regarding log message

**Review by haggaret — APPROVED** [review]
See comments :)

**Pull request merged**